### PR TITLE
Update tests to `lxml > 4.7`

### LINKED ISSUE #107: Tests break with lxml >= 4.7
The only difference between 
* https://github.com/zopefoundation/z3c.form/actions/runs/1569472646 (success)
* https://github.com/zopefoundation/z3c.form/actions/runs/1598508175 (failure in all Python versions)

is `lxml==4.6.4` => `lxml==4.7.1`

### PR DESCRIPTION
**Pull request by icemac**
Requiring at least that version.

Fixes #107.

### CODE REVIEW

File: src/z3c/form/form.rst [hunk 1]
```diff
@@ -922,12 +922,12 @@ When the form renders, the label has now changed:
   >>> addForm = PersonAddForm(root, TestRequest())
   >>> addTemplate(addForm)
   >>> addForm.update()
-  >>> print(testing.render(addForm, './/xmlns:div[2][@class="row"]'))
+  >>> print(testing.render(addForm, './/xmlns:div[2][@class="row"]'))  # doctest: +NOPARSE_MARKUP
```
Comment (jamadden) [inline src/z3c/form/form.rst:925]: Could you comment on what `NOPARSE_MARKUP` is meant to be doing? I'm not familiar with that flag, and I don't see it listed as a flag in [the documentation](https://docs.python.org/3/library/doctest.html#option-flags), nor have I been able to find it any any other zopefoundation repos.

**Review by jamadden — COMMENTED** [review]
This looks like it's just differences in pretty-printing, right? 

Because:

1) The pin is only for tests;
2) The pin doesn't drop any supported versions;
3) The pin actually brings in a number of important CVE security fixes

I'm totally fine with updating the pin.

File: src/z3c/form/form.rst (hunk 1 above)
Comment (icemac) [reply src/z3c/form/form.rst:925]: `NOPARSE_MARKUP` is a custom flag of `z3c.form`, it is registered here:  https://github.com/zopefoundation/z3c.form/blob/440eb18c541b88f64e88be9b66fbc27748f75af2/src/z3c/form/outputchecker.py#L44

It is used to select the parser for the markup:

https://github.com/zopefoundation/z3c.form/blob/440eb18c541b88f64e88be9b66fbc27748f75af2/src/z3c/form/outputchecker.py#L98-L108

As the default parser now returns the HTML tail which follows the `xpath` selection not parsing the markup helps the tests to succeed.

**Comment by icemac** [comment]
Thank you for looking into this PR. 😃

**Pull request merged**